nucypher/blockchain/eth/actors.py

```diff
@@ -1194,15 +1198,80 @@ def __init__(self,
 
         # Workers cannot be started without being assigned a stake first.
         if is_me:
+            if check_active_worker:
+                self._checksum_address = self.block_until_ready(worker_address, self.registry)
             self.stakes = StakeList(registry=self.registry, checksum_address=self.checksum_address)
             self.stakes.refresh()
-            if check_active_worker and not len(self.stakes):
-                raise self.DetachedWorker(f"{self.__worker_address} is not bonded to {self.checksum_address}.")
 
             self.work_tracker = work_tracker or WorkTracker(worker=self)
             if start_working_now:
                 self.work_tracker.start(act_now=False)
 
+    @classmethod
+    def block_until_ready(cls,
+                          worker_address: str,
+                          registry: 'BaseContractRegistry',
+                          poll_rate: int = None,
+                          timeout: int = None) -> str:
+        """
+        Polls the staking_agent and blocks until the staking address is not
+        a null address for the given worker_address.
+        Once the worker is bonded, it returns the staker address.
+        """
+
+        timeout = timeout or cls.BONDING_TIMEOUT
+        poll_rate = poll_rate or cls.BONDING_POLL_RATE
+        staking_agent = ContractAgency.get_agent(StakingEscrowAgent, registry=registry)
+        client = staking_agent.blockchain.client
+        start = maya.now()
+        emitter = StdoutEmitter()  # TODO: Make injectable, or embed this logic into Ursula
+
+        counter, funded, bonded = 0, False, False
+        while True:
+
+            # Read
+            staking_address = staking_agent.get_staker_from_worker(worker_address)
+            ether_balance = client.get_balance(worker_address)
+
+            # Bonding
+            if staking_address == BlockchainInterface.NULL_ADDRESS:
+                if counter % 10 == 0:
+                    emitter.message("Waiting for bonding...", color="yellow")
+            else:
+                if not bonded:
+                    bonded = True
+                    emitter.message(f"Worker is bonded to ({staking_address})!", color='green', bold=True)
+
+            # Balance
+            if not ether_balance:
+                if counter % 10 == 0:
```

Comment:
```suggestion
            if not ether_balance and counter % 10 == 0
```

Dunno, I just think condensing this logic makes this way more readable from my perspective.

Reply:
:heavy_check_mark: